### DISCUSSION TITLE
Get branch name

### DISCUSSION THREAD

**Discussion by reikdas** [community/community#25363] [Actions]
<p>For context, this is my Github Actions ci.yml - <a href="https://github.com/reikdas/cling/blob/e9e6d80e3f15568d05a869ab1e314f453a857b96/.github/workflows/ci.yml" rel="nofollow noopener">https://github.com/reikdas/cling/blob/e9e6d80e3f15568d05a869ab1e314f453a857b96/.github/workflows/ci.yml</a></p>
<p>I want to get the name of the branch that is creating a PR or the name of the branch a commit is pushed to (when Github Actions is triggered).</p>
<p>I found separate solutions for PR branch and commit branch.<br>
For PR branch I found this - <a href="https://github.community/t/how-to-get-pr-branch-name-in-github-actions/16598/8" class="inline-onebox">How to get PR branch name in GitHub Actions</a><br>
and for commit branch I got this - <a href="https://stackoverflow.com/a/58035262" rel="nofollow noopener">https://stackoverflow.com/a/58035262</a> and several others which state the same thing.</p>
<p>Do I need to create separate steps for a pull request and a push? I don’t care about triggering both at the same time since I want push to be triggered only on master and pull request only on branches to master (as specified in the above linked ci.yml); so the two of them will never be triggered together.</p>
<p>I would really appreciate any help with this :slight_smile:</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<aside class="quote no-group" data-username="reikdas" data-post="1" data-topic="117644">
<div class="title">
<div class="quote-controls"></div>
 reikdas:</div>
<blockquote>
<p>Do I need to create separate steps for a pull request and a push?</p>
</blockquote>
</aside>
<p>My answer is yes. As the two events <a href="https://developer.github.com/webhooks/event-payloads/#push" rel="nofollow noopener">push</a> and <a href="https://developer.github.com/webhooks/event-payloads/#pull_request" rel="nofollow noopener">pull_request</a> have different event payload , I would recommend you add two steps to get PR branch and push commit branch. You could add if conditional to these two steps:</p>
<pre><code class="lang-auto">if: github.event_name == 'pull_request'
</code></pre>
<p>and</p>
<pre><code class="lang-auto">if: github.event_name == 'push'
</code></pre>
